doc/source/train/user-guides/data-loading-preprocessing.rst

```diff
@@ -177,25 +185,123 @@ Using Ray Data and Ray Train for distributed training on large datasets involves
+
+.. _train-datasets-pytorch:
+
+Starting with PyTorch data
+--------------------------
+
+Some frameworks provide their own dataset and data loading utilities. For example:
+
+- **PyTorch:** `Dataset & DataLoader <https://pytorch.org/tutorials/beginner/basics/data_tutorial.html>`_
+- **Hugging Face:** `Dataset <https://huggingface.co/docs/datasets/index>`_
+- **PyTorch Lightning:** `LightningDataModule <https://lightning.ai/docs/pytorch/stable/data/datamodule.html>`_
+
+These utilities can still be used directly with Ray Train. In particular, you may want to do this if you already have your data ingestion pipeline set up.
+However, for more performant large-scale data ingestion we do recommend migrating to Ray Data.
+
+At a high level, you can compare these concepts as follows:
+
+.. list-table::
+   :header-rows: 1
+
+   * - PyTorch API
+     - HuggingFace API
+     - Ray Data API
+   * - `torch.utils.data.Dataset <https://pytorch.org/docs/stable/data.html#torch.utils.data.Dataset>`_
+     - `datasets.Dataset <https://huggingface.co/docs/datasets/main/en/package_reference/main_classes#datasets.Dataset>`_
+     - :class:`ray.data.Dataset`
+   * - `torch.utils.data.DataLoader <https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader>`_
+     - n/a
+     - :meth:`ray.data.Dataset.iter_torch_batches`
+
+
+For more details, see the following sections for each framework.
+
+.. tabs::
+
+    .. tab:: PyTorch Dataset and DataLoader
+
+        **Option 1:** Convert your PyTorch Dataset to a Ray Dataset and pass it into the Trainer via  ``datasets`` argument.
+        Inside your ``train_loop_per_worker``, you can access the dataset via :meth:`ray.train.get_dataset_shard`. 
+        You can convert this to replace the PyTorch DataLoader via :meth:`ray.data.DataIterator.iter_torch_batches`.
+         
+        For more details, see the :ref:`Migrating from PyTorch Datasets and DataLoaders <migrate_pytorch>`.
 
-For more details, see the :ref:`Ray Data PyTorch guide <migrate_pytorch>` and :ref:`Ray Data for HuggingFace and TensorFlow <loading_datasets_from_ml_libraries>`.
+        **Option 2:** Instantiate the Torch Dataset and DataLoader directly in the ``train_loop_per_worker``.
+        You can use the :meth:`ray.train.torch.prepare_data_loader` utility to set up the DataLoader for distributed training.
 
-.. _train_datasets_configuration:
+    .. tab:: Hugging Face Dataset
 
-Customizing how to split datasets
----------------------------------
+        **Option 1:** Convert your Hugging Face Dataset to a Ray Dataset and pass it into the Trainer via the ``datasets`` argument.
+        Inside your ``train_loop_per_worker``, you can access the dataset via :meth:`ray.train.get_dataset_shard`. 
+
+        For instructions, see :ref:`Ray Data for Hugging Face <loading_datasets_from_ml_libraries>`.
+
+        **Option 2:** Instantiate the Hugging Face Dataset directly in the ``train_loop_per_worker``.
+
+    .. tab:: LightningDataModule
```

Comment:
Can we move lightning to the second tab, so that the users know the "same logic" here is in the first tab (PyTorch Dataset and DataLoader)?